packages/flutter_tools/lib/src/commands/migrate_resolve_conflicts.dart

```diff
@@ -0,0 +1,322 @@
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory(logger: logger, fileSystem: fileSystem);
+    final FlutterProject project = projectDirectory == null
+      ? FlutterProject.current()
+      : flutterProjectFactory.fromDirectory(fileSystem.directory(projectDirectory));
+
+    Directory stagingDirectory = project.directory.childDirectory(kDefaultMigrateStagingDirectoryName);
+    final String? customStagingDirectoryPath = stringArg('staging-directory');
+    if (customStagingDirectoryPath != null) {
+      if (fileSystem.path.isAbsolute(customStagingDirectoryPath)) {
+        stagingDirectory = fileSystem.directory(customStagingDirectoryPath);
+      } else {
+        stagingDirectory = project.directory.childDirectory(customStagingDirectoryPath);
+      }
+    }
+    if (!stagingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('flutter migrate start', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile = MigrateManifest.getManifestFileFromDirectory(stagingDirectory);
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+
+    checkAndPrintMigrateStatus(manifest, stagingDirectory, logger: logger);
+
+    final List<String> conflictFiles = manifest.remainingConflictFiles(stagingDirectory);
+
+    terminal.usesTerminalUi = true;
+
+    for (int i = 0; i < conflictFiles.length; i++) {
+      final String localPath = conflictFiles[i];
+      final File file = stagingDirectory.childFile(localPath);
+      final List<String> lines = file.readAsStringSync().split('\n');
+      // We write a newline in the output, this counteracts it.
+      if (lines.last == '') {
+        lines.removeLast();
+      }
+
+      // Find all conflicts
+      final List<Conflict> conflicts = findConflicts(lines);
+
+      // Prompt developer
+      await promptDeveloperSelectAction(conflicts, lines, localPath);
+
+      final bool result = await verifyAndCommit(conflicts, lines, file, localPath);
+      if (!result) {
+        i--;
+      }
+    }
+    return const FlutterCommandResult(ExitStatus.success);
+  }
+
+  /// Parses the lines of a file and extracts a list of Conflicts.
+  List<Conflict> findConflicts(List<String> lines) {
+    // Find all conflicts
+    final List<Conflict> conflicts = <Conflict>[];
+    Conflict currentConflict = Conflict.empty();
+    for (int lineNumber = 0; lineNumber < lines.length; lineNumber++) {
+      final String line = lines[lineNumber];
+      if (line.contains(_conflictStartMarker)) {
+        currentConflict.startLine = lineNumber;
+      } else if (line.contains(_conflictDividerMarker)) {
+        currentConflict.dividerLine = lineNumber;
+      } else if (line.contains(_conflictEndMarker)) {
+        currentConflict.endLine = lineNumber;
+        assert(
```

Comment:
can this be a runtime check that throws a `StateError` instead of an assert? asserts only run in unit tests, and I think we should surface this in crash reporting if we ever hit it.